std/datetime/date.d

```diff
@@ -7549,40 +7549,36 @@ public:
     static Date fromISOExtString(S)(in S isoExtString) @safe pure
         if (isSomeString!(S))
     {
-        import std.algorithm.searching : all, startsWith;
-        import std.ascii : isDigit;
-        import std.conv : to;
-        import std.exception : enforce;
+        import std.algorithm.searching : startsWith;
+        import std.conv : to, ConvException;
         import std.format : format;
         import std.string : strip;
 
-        auto dstr = to!dstring(strip(isoExtString));
-
-        enforce(dstr.length >= 10, new DateTimeException(format("Invalid ISO Extended String: %s", isoExtString)));
+        auto str = strip(isoExtString);
+        short year;
+        ubyte month, day;
 
-        auto day = dstr[$-2 .. $];
-        auto month = dstr[$-5 .. $-3];
-        auto year = dstr[0 .. $-6];
+        if (str.length < 10 || str[$-3] != '-' || str[$-6] != '-')
+            throw new DateTimeException(format("Invalid ISO Extended String: %s", isoExtString));
 
-        enforce(dstr[$-3] == '-', new DateTimeException(format("Invalid ISO Extended String: %s", isoExtString)));
-        enforce(dstr[$-6] == '-', new DateTimeException(format("Invalid ISO Extended String: %s", isoExtString)));
-        enforce(all!isDigit(day),
-                new DateTimeException(format("Invalid ISO Extended String: %s", isoExtString)));
-        enforce(all!isDigit(month),
-                new DateTimeException(format("Invalid ISO Extended String: %s", isoExtString)));
+        auto yearStr = str[0 .. $-6];
+        auto signAtBegining = yearStr.startsWith('-', '+');
+        if ((yearStr.length > 4 && !signAtBegining) ||
+            (yearStr.length <= 4 && signAtBegining))
+            throw new DateTimeException(format("Invalid ISO Extended String: %s", isoExtString));
```

Comment:
I hate to nitpick style, but would you please use braces if the condition is more than one line - either that or put the condition on a single line instead of breaking it up. IMHO, it's a readability problem if there are no braces when the condition is multiple lines.

Reply:
Also, aside from the optimizer being smart about things here (which it may be), I would expect it to be faster if you broke this up so that it didn't check the length twice given that it's the same check in reverse.